util/text/text_test.go

```diff
@@ -43,7 +44,7 @@ func TestWrap(t *testing.T) {
 		// A tab counts as 4 characters.
 		{
 			"foo\nb\t r\n baz",
-			"foo\nb\n  r\n baz",
+			"foo\nb\nr\n baz",
```

Comment:
The idea here was that a `\t` character is an explicit 4 spaces and should probably not be trimmed or reduced to a single space. In other words, the algorithm should trim spaces only after the wrapping as been done.

Reply:
If I had to choose I would keep the previous behavior here (the tabulation spacing is preserved across lines), but your behavior is not really wrong either. If you can make it works, that's good, but it's fine otherwise.

Reply:
I can try to preserve the previous behavior. But I don't quite understand the rule here. `b\t r\n` results in `b\n  r\n`, which means the 4 spaces are trimmed in a specific way.

Reply:
The idea is that the tabulation is replaced by a 4 character long space and keep providing a constant spacing. I didn't try with your algorithm, but my guess is that if the tabulation is at the end of the line, it will be trimmed and disappear. The previous behavior is that the next line would be pushed with an extra padding to keep this spacing.

Reply:
Sorry for not getting the idea. Why 'b\tSr\n' results in `b\nSSr\n` given a 4-char limit? Shouldn't it be `b\nSSSS\nSr\n`?

Reply:
Another thing I failed to understand is the line https://github.com/MichaelMure/git-bug/blob/433c4fc46a7a2cabb69eee5226bb8ea87b6c5661/util/text/text.go#L26, which substitute tabs for 4 spaces, but only for the first 4 occurrences? But if tabs are substituted before processing, how do we know when we shall treat the tabs in a special way?

Reply:
Ho well, you are right, the previous code was buggy. 

> Sorry for not getting the idea. Why 'b\tSr\n' results in `b\nSSr\n` given a 4-char limit? Shouldn't it be `b\nSSSS\nSr\n`?

You are correct. That said, this behavior is more a wish than a requirement. I think that would look better, but if you can't make it work, it's fine.